### ISSUE TITLE
plot.ctd() maps do not always include sufficient nearby land

### ISSUE THREAD

**Issue by dankelley** [dankelley/oce#828]
For example, the code for #816 required a `span` argument to show what I'd regard as sufficient land. I've seen this before, quite a lot.

**Comment by dankelley** [comment]
As an example, see the following code and graph.
```r
library(oce)
##> try(source("~/src/oce/R/ctd.R"))
##> try(source("~/src/oce/R/coastline.R"))
data(ctd)
Hlon <- -157.8
Hlat <- 21.2 - 0.2
ctd[["longitude"]] <- Hlon
ctd[["latitude"]] <- Hlat
if (!interactive()) png("818a.png")
par(mfrow=c(2,2))
for (dlon in c(-0.2, 0.2)) {
    for (dlat in c(-0.2, 0.2)) {
        ctd[["latitude"]] <- Hlat + dlat
        ctd[["longitude"]] <- Hlon + dlon
        plot(ctd, which='map')
    }
}
if (!interactive()) dev.off()
```

![818a](https://cloud.githubusercontent.com/assets/99469/12362779/88c980f8-bb9b-11e5-9cca-21c8359c14f7.png)

**Comment by dankelley** [comment]
I tweaked the code (not pushed) to set scale to 3X nearest land and got as below. So, it's a bit better (on this individual case.)

But there is another issue: as a ship moves around, the auto-scale will change. That can be quite distracting when making plots into a movie.

I think we should have some standardization, e.g. maybe if nearest land is between 50 and 100km, set to 100km, etc. What do you think, @richardsc ?

![818a](https://cloud.githubusercontent.com/assets/99469/12362990/a1204014-bb9c-11e5-8529-a939a8d91602.png)

**Comment by richardsc** [comment]
> @dankelley: But there is another issue: as a ship moves around, the auto-scale will change. That can be quite distracting when making plots into a movie.

Yes, but if the plan is to make plots into a movie I don't think it's unreasonable to set the `span` explicitly. If I were doing this I would probably choose whichever span best suited the *entire* dataset, rather than rely on the auto-spanning.

I think the issue as raised is addressed, and I would vote to not dive too deep into edge cases (for once! :P)

**Comment by dankelley** [comment]
Cool, closing. Things can always be adjusted. I put a note in the docs about how this slows things down (it has to compute distances to every point in the world coastline, and then -- potentially very expensively -- it has to order them)

**Comment by dankelley** [comment]
Yeah, well, I like edge cases :smile: so I coded in a new fcn called `bound125()` and use that now. I just find it terribly distracting to page through plots and see the scale changing on autoscaled plots. In a lot of real cases I end up setting `span` myself though, so I agree on that. There is no perfect way to do this sort of thing. I dropped my previous method of finding the 20 nearest points since that is computationally wasteful, and now I have `bound125()` of a multiple of the nearest point. (I assume `which.min()` is cheap; it's an O(n) calculation, after all, and R things tend to be reasonably tightly coded.)
